internal/sys/signals.go

```diff
@@ -0,0 +1,41 @@
+package sys
+
+import (
+	"fmt"
+	"runtime"
+
+	"github.com/cilium/ebpf/internal/unix"
+)
+
+var profSet unix.Sigset_t
+
+func init() {
+	if err := unix.SigsetAdd(&profSet, unix.SIGPROF); err != nil {
+		panic(fmt.Errorf("creating signal set: %w", err))
+	}
+}
+
+// MaskProfilerSignal locks the calling goroutine to its underlying OS thread
+// and adds SIGPROF to the thread's signal mask. This prevents pprof from
+// interrupting expensive syscalls like e.g. BPF_PROG_LOAD.
+//
+// Call defer sys.UnmaskProfilerSignal() to reverse the operation.
+func MaskProfilerSignal() {
+	runtime.LockOSThread()
```

Comment:
Just to be on the safe side - we don't create any goroutines in between `runtime.LockOSThread()` and `runtime.UnlockOSThread()`? Asking, as such a goroutine might end up running on a thread with a non-modified sigmask.

Reply:
No worries, we are aware of this!